docs/demo.rst

```diff
@@ -0,0 +1,22 @@
+A Simple Demo for GluonNLP
```

Comment:
We may have more than 1 demo. We should have a "demo" category, and then with a drop down list with a few different demos:
- sentiment analysis
- question answering